cpp/cmake/TileDB.cmake

```diff
@@ -9,7 +9,18 @@ message(STATUS "start to set tiledb for version:${TILEDB_VERSION}")
 if(${TILEDB_VERSION} MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+)$")
 message(STATUS "start to set TILEDB_DOWNLOAD_URL and TILEDB_DOWNLOAD_SHA1")
 
-if(${TILEDB_VERSION} STREQUAL "2.9.4")
+if(${TILEDB_VERSION} STREQUAL "2.10.1")
```

Comment:
Why do we do matching on the version?

Reply:
This was carried over from the earlier API, there's definitions for the download targets for several versions below.